sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -586,6 +628,40 @@ public void setValueDeserializer(String valueDeserializer) {
       return toBuilder().setValueDeserializer(valueDeserializer).setValueCoder(valueCoder).build();
     }
 
+    /**
+     * Allows to specify custom {@link CSRClientProvider}. {@link CSRClientProvider} provides {@link
+     * SchemaRegistryClient} instance which is used later for communication with Confluent Schema
+     * Registry. You should use this method if {@link Read#withCSRClientProvider(String, String,
+     * String)} does not suit your needs.
+     */
+    public Read<K, V> withCSRClientProvider(CSRClientProvider csrClientProvider) {
+      Builder<K, V> builder = toBuilder();
+      if (csrClientProvider.getKeySchemaSubject() != null) {
+        builder.setKeyDeserializer((Class) KafkaAvroDeserializer.class);
+      }
+      if (csrClientProvider.getValueSchemaSubject() != null) {
+        builder.setValueDeserializer((Class) KafkaAvroDeserializer.class);
+      }
+      return builder
+          .setCSRClientProvider(csrClientProvider)
+          .build()
+          .withConsumerConfigUpdates(
+              ImmutableMap.of(
+                  KafkaAvroDeserializerConfig.SCHEMA_REGISTRY_URL_CONFIG,
+                  csrClientProvider.getSchemaRegistryUrl()));
+    }
+
+    /**
+     * Specify Confluent schema registry url, key and value schema subjects to be used to read
+     * schema from Confluent Schema Registry. If you need more sophisticated credential protocol,
```

Comment:
Remove the comment `"If you need more sophisticated version..."` for consistency with non public exposure of the related method.